src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs

```diff
@@ -99,7 +92,8 @@ public static TokenCredential GetTokenCredential(string resource, string tenantI
                     tokenCredential = new ClientSecretCredential(tenantId, authenticationInfo.IdentityId, authenticationInfo.Secret, tokenCredentialOptions);
                     break;
                 case AuthenticationMode.UserAssignedManagedIdentity:
-                    tokenCredential = new ManagedIdentityCredential(authenticationInfo.IdentityId, tokenCredentialOptions);
+                    var clientId = string.IsNullOrWhiteSpace(authenticationInfo.IdentityId) ? null : authenticationInfo.IdentityId;
+                    tokenCredential = new ManagedIdentityCredential(clientId, tokenCredentialOptions);
```

Comment:
I presume this does not complain in case it is null?

Reply:
It does not as null is the default value of the argument